app/models/manageiq/providers/microsoft/infra_manager/refresh_parser.rb

```diff
@@ -330,6 +330,8 @@ def process_vm_hardware(vm)
     end
 
     def process_snapshots(p)
+      return unless p[:VMCheckpoints]
```

Comment:
@chessbyte 
Both an empty array and NIL will work, the VM[:snapshots] hash is not processed when it is set to NIL:
https://github.com/ManageIQ/manageiq/blob/master/app/models/ems_refresh/save_inventory.rb#L296

(BTW, the long silence was because I've been on PTO from last Friday through today inclusive!)